Add Sample Testplans to Documentation (#37)
yahoo/ychaos: docs/testplan/sample_test_plan.md

> This section of the document contains some example testplans for you to get started quickly on the project

Reviewer: if you get time, can you also do a quick validation of the test plans here using cli? ignore if already done.

#### CPU Burn
```yaml

> description: Increase CPU utlization on all the cores for 60 second

Reviewer: ```suggestion
description: Increase CPU utilisation on all the cores for 60 seconds
```

attack:
  target_type: machine
  target_config:
    blast_radius: 100
    ssh_config:
      user: testUser
      password: testUserPassword
    hostnames:
    - mocktargethost.namespace.cloud
    report_dir: "./"
  agents:
    - type: cpu_burn
      config:
        start_delay: 10 # Start Burn CPU with 10s delay
        duration: 60 # Burn CPU for 60s
```

#### Disable Ping
```yaml
...
description: Disable ping response for 60 second
agents:
  - type: ping_disable
    config:
      duration: 60
      start_delay: 0
```

#### Block Ports 
```yaml
...

> description: Block inbound and outbound connections on port 80 and 8080 for 60 second

Reviewer: ```suggestion
description: Block inbound and outbound connections on port 80 and 8080 for 60 seconds
```

agents:
  - type: block_ports
    config:
      start_delay: 0
      duration: 60
      incoming_ports:
        - 80
        - 8080
      destination_ports:
        - 80
        - 8080
```

#### Block Endpoints 
```yaml
...
description: Block incoming and outgoing connections for the hostnames for 60 second
agents:
  - type: block_ports
    config:
      start_delay: 0
      duration: 60
      incoming_endpoints:
        - "203.0.113.0"
        - "https://yahoo.com:443"
      outgoing_endpoints:
        - "203.0.113.0"
        - "https://yahoo.com:443"
```

#### Block DNS Requests
```yaml
...
description: Block DNS requests for 60 second
agents:
  - type: block_dns
    config:
      start_delay: 0
      duration: 60
```

#### Block Traffic
```yaml
...
description: Block Requests to hostnames for 60 second
agents:
  - type: traffic_block
    config:
      start_delay: 0
      duration: 60
      hosts:
        - "https://api.screwdriver.cd"
        - "https://api.nationalize.io"
```

#### Endpoint Certificate Validation
```yaml
...
description: Verify endpoint cert expire date
agents:

>   - type: urls

Reviewer: should be server_cert_validation ?
```suggestion
  - type: server_cert_validation
```

    config:
      expiry_threshold: 7 # check if cert expire date is more that 7days
      urls:
        - "www.techcrunch.com"
        - "www.aol.com"
```
!!! note ""
    For more information on attack agents [refer](/ychaos/agents/)

#### HTTP Request Verification Plugin
```yaml
...

> description: Verify Steady state by checking Request latency is bellow 100ms

Reviewer: ```suggestion
description: Verify Steady state by checking Request latency is below 100ms
```

verification:
  - delay_before: 10
  - state: 'STEADY'
  - type: 'http_request'
  - config:
      latency: 100
      urls:
        - 'api.nationalize.io'
      method: 'GET'
      params:
        name: 'india'
      status_codes:
        - 200
      count: 10
attack:
    target_type: machine
    agents:
        - type: no_op
```

#### Screwdriver V4 Verification Plugin
```yaml
...

> description: Use Screwdrive job to verify Steady state   

Reviewer: ```suggestion
description: Use Screwdriver job to verify Steady state   
```

> #### CPU Burn

Reviewer 2: Add a header note 
```
This section of the document contains some example testplans for you to get started quickly on the project
```

Reviewer 2: And how about renaming the file to `examples.md` and the page title to "Test Plan Examples"

```yaml
attack:
  target_type: machine
  target_config:
    blast_radius: 100
    ssh_config:
      user: testUser
      password: testUserPassword
    hostnames:
    - mocktargethost.namespace.cloud
    report_dir: "./"
  agents:
    - type: cpu_burn
      config:
        start_delay: 10 # Start Burn CPU with 10s delay
        duration: 60 # Burn CPU for 60s
```

#### Disable Ping
```yaml
...
agents:
  - type: ping_disable
    config:
      duration: 60
      start_delay: 0
```

#### Traffic Block
```yaml
...
agents:
  - type: traffic_block
    config:
      duration: 60
      start_delay: 0
      hosts:
        - 'www.yahoo.com'

>         - 'www.google.com'

Reviewer 2: Lets use some real time example of APIs for these like AWS/Redis APIs

```

#### Block Ports 
```yaml
...
agents:
  - type: block_ports
    config:
      start_delay: 0
      duration: 60
      incoming_ports:
        - 80
        - 8080
      destination_ports:
        - 80
        - 8080
```

#### Block Endpoints 
```yaml
...
agents:
  - type: block_ports
    config:
      start_delay: 0
      duration: 60
      incoming_endpoints:
        - "203.0.113.0"
        - "https://yahoo.com:443"
      outgoing_endpoints:
        - "203.0.113.0"
        - "https://yahoo.com:443"
```

#### Block DNS Requests
```yaml
...
agents:
  - type: block_dns
    config:
      start_delay: 0
      duration: 60
```

#### Block Traffic
```yaml
...
agents:
  - type: traffic_block
    config:
      start_delay: 0
      duration: 60
      hosts:
        - "yahoo.com"
        - "google.com"

> ```

Reviewer 2: Repeated?

Author: yes.. removed

#### Endpoint Certificate Validation
```yaml
...
agents:
  - type: urls
    config:
      expiry_threshold: 7 # check if cert expire date is more that 7days
      urls:
        - "www.yahoo.com"

>         - "www.google.com"

Reviewer 2: Lets use some URLs from Yahoo only (www.aol.com, www.techcrunch.com)

```
!!! note ""
    For more information on attack agents [refer](/ychaos/agents/)

#### HTTP Request Verification Plugin
```yaml
...
verification:
  - delay_before: 10
  - state: 'STEADY'
  - type: 'http_request'
  - config:
      latency: 100
      urls:
        - 'api.nationalize.io'
      method: 'GET'
      params:

>         name: 'india'

Reviewer 2: Just want to know is there an API like this? No need to change but just for information

Author: yes there is a API `https://api.nationalize.io/?name=india`

      status_codes:
        - 200
      count: 10
attack:
    target_type: machine
    agents:
        - type: no_op
```

#### Screwdriver V4 Verification Plugin
```yaml
...
verification:
  - delay_before: 10
  - state: 'STEADY'
  - type: 'sdv4'
  - config:
      sd_api_url: 'api.screwdriver.cd'
      pipeline_id: 123123 # SD pipeline id
      job_name: 'Run_FT' # SD job name

>       sd_api_token: 'API token' # Refer Screwdriver Docs

Reviewer 2: Lets use the environment variable type of secret here.
```suggestion
      sd_api_token: # Refer Screwdriver Docs
          type : env
          id: SD_API_TOKEN
```

attack:
    target_type: machine
    agents:
        - type: no_op
```
!!! note ""
    For more information on verification plugins [refer](/ychaos/verification/)

> 

Reviewer 2: nit: Extra Line
